code/datums/components/beacon.dm

```diff
@@ -159,8 +165,36 @@
 	activator = null
 	playsound(source, 'sound/machines/twobeep.ogg', 15, 1)
 	active = FALSE //this is here because of attack hand
+	UnregisterSignal(source,COMSIG_MOVABLE_MOVED)
+	if(source.loc)
+		UnregisterSignal(source.loc,COMSIG_MOVABLE_MOVED)
 	source.update_appearance()
 
+///Updates position and name of the beacon, or alternatively turns if off if it's in a blacklisted area.
+/datum/component/beacon/proc/updatepos(atom/movable/source)
+	SIGNAL_HANDLER
+	var/turf/location = get_turf(parent)
+	var/area/A = get_area(location)
+	if(A && istype(A) && A.ceiling >= CEILING_DEEP_UNDERGROUND)
```

Comment:
```suggestion
	if(istype(A) && A.ceiling >= CEILING_DEEP_UNDERGROUND)
```
The istype check should account for it being null, I'm also not sure when get_area would ever return anything that isn't an area.